frontend/app/src/scenes/Home/scenes/UpsertCategory/index.tsx

```diff
@@ -103,6 +134,38 @@ class UpsertCategory extends React.Component<
       });
   };
 
+  onAddTranslation = () => {
+    this.setState({
+      isInLanguageSelectionMode: true,
+    });
+  };
+
+  // Checks for the selected language to render the relevant language card
+  onLanguageSelect = (language: string) => {
+    const clonedTranslations = new Map(this.state.translations);
+    clonedTranslations.set(language, { name: '' });
+    this.setState({
+      isInLanguageSelectionMode: false,
+      translations: clonedTranslations,
+    });
+  };
+
+  handleTranslations = (languageEnum: string, value: string) => {
```

Comment:
what about, `onTranslationEdit` or `onEditTranslation`?

Reply:
it's not edit, it adding

Reply:
it calls whenever you change the value of the field right?

Reply:
yeah